src/main/java/ch/uzh/ifi/seal/soprafs20/controller/UserController.java

```diff
@@ -53,4 +55,52 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
         // convert internal representation of user back to API
         return DTOMapper.INSTANCE.convertEntityToUserGetDTO(createdUser);
     }
+
+    @GetMapping("/user/{userId}")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public UserGetDTO getUserByUserName(@RequestHeader("X-Auth-Token") String token, @PathVariable Long userId){
+        UserGetDTO userGetDTO = new UserGetDTO();
+        return  userGetDTO;
+
+    }
+
+    @GetMapping("/user/login")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @ResponseBody
+    public UserAuthDTO login(@RequestBody UserPostDTO userPostDTO){
+        UserAuthDTO userAuthDTO = new UserAuthDTO("supersecrettokenvalue");
+        return userAuthDTO;
+    }
+
+    @PutMapping("/logout/{userId}")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void logout(@RequestHeader("X-Auth-Token") String token,@PathVariable Long userId){
+        return;
+    }
+
+    @PutMapping("/user/{userId}/edit")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public UserUpdateDTO updateUser(@RequestHeader("X-Auth-Token") String token, @RequestBody User user, @PathVariable Long userId){
+        UserUpdateDTO userUpdateDTO = new UserUpdateDTO();
+        return userUpdateDTO;
+    }
+
+    @PutMapping("/user/{userId}/password")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public void updatePassword(@RequestHeader("X-Auth-Token") String token,@RequestBody UserPostDTO userPostDTO){
+        return;
+    }
+
+    @PutMapping("/user/{userId}/invitation")
+    public List<Long> invitation(@RequestHeader("X-Auth-Token") String token,@PathVariable Long userId){
```

Comment:
The request body is missing here, in the class diagram I am using a UserPUTDTO, but up to you, if you want to use a separate one again. There also no need to return anything here, but be sure to set the response status to OK.